src/aws.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @typedef {object} ParsedTopic
+ * @property {string|undefined} region - aws region of topic
+ */
+
+/**
+ * Parse an AWS SNS TopicArn
+ * @param {string} topicArn - e.g. arn:aws:sns:us-east-2:foo:topicName
+ * @returns {ParsedTopic}
+ */
+function parseTopic(topicArn) {
+  const pattern = /arn:aws:sns:([^:]*)/i
```

Comment:
Should the string start with this or can be anywhere?

```suggestion
  const pattern = /^arn:aws:sns:([^:]*)/i
```